api/src/test/java/org/openmrs/module/outgoingmessageexceptions/api/OutgoingMessageExceptionsServiceTest.java

```diff
@@ -81,17 +81,29 @@ public void saveItem_shouldSetOwnerIfNotSet() {
 	}
 	
 	@Test
-	public void shouldReturnMessageById() throws Exception {
+	public void shouldReturnSerializedMessageById() throws Exception {
 		Integer testId = 1;
 		
 		when(dao.getMessageById(1)).thenReturn(prepareDummyOutgoingMessage());
 		
 		String expected = readJsonFromFile(OUTGOING_MESSAGE_RESPONSE_JSON);
-		String fetched = basicModuleService.getMessageById(testId);
+		String fetched = basicModuleService.getSerializedMessageById(testId);
 		
 		assertEquals(expected, fetched);
 	}
 	
+	@Test
+	public void shouldReturnMessageById() {
+		Integer testId = 1;
+		OutgoingMessage dummyOutgoingMessage = prepareDummyOutgoingMessage();
+		
+		when(dao.getMessageById(1)).thenReturn(dummyOutgoingMessage);
+		
+		OutgoingMessage fetched = basicModuleService.getMessageById(testId);
+		
+		assertEquals(dummyOutgoingMessage, fetched);
```

Comment:
For correct work you need to override the equals and hashcode methods.

Reply:
Corrected in ITOS-32 PR